### PR TITLE
Integ online

### PR DESCRIPTION
**Pull request by ti2ger92**
first pass

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - uses: actions/checkout@v2
+      with:
+        repository: 'upchieve/server'
+        path: 'server'
+    # Checks-out this web repository under $GITHUB_WORKSPACE/web
+    - uses: actions/checkout@v2
+      with:
+        path: 'web'
+    # Create an empty directory to use for the volume, and start the container
+    - run: cd server && cp config.example.js config.js && mkdir mongo-volume && docker-compose up -d
```
Comment (dmnisson) [inline .github/workflows/e2e_tests.yml:33]: The example config file doesn't have any volunteer codes set by default, but the e2e tests need to sign up volunteers with valid codes in order to pass. We'll need to find a way to get these tests to run without revealing production signup codes to the public.

Comment (jkeat) [reply .github/workflows/e2e_tests.yml:33]: We can just add some default (fake) volunteer codes, right?

**Comment by jkeat** [comment]
@jkeat get tests passing or disable them

**Comment by ti2ger92** [comment]
after a merge from master, this is now dependent on https://github.com/UPchieve/server/pull/356 .

**Comment by jkeat** [comment]
Getting close!
Screen Shot 2020-05-14 at 5 18 23 PM

**Comment by ti2ger92** [comment]
impressively done, nice!

File: package-lock.json [hunk 2]
```diff
@@ -5,641 +5,646 @@
   "requires": true,
```
Comment (jkeat) [inline package-lock.json:5]: Was this what npm generated from a fresh install or something? Trying to understand what led to so many minor updates throughout the lockfile when there wasn't anything added to package.json

Comment (jkeat) [reply package-lock.json:5]: I think that's why the linting rules changed slightly (I had to reformat some stuff to get the check to pass). In theory, minor version changes shouldn't break stuff, but this is a _lot_ at once.

Comment (jkeat) [reply package-lock.json:5]: Unless there's some essential update in `package-lock.json`, I'm gonna revert that part of the PR (lock file changes + the lint changes it caused)

**Pull request merged**